src/body.rs

```diff
@@ -175,6 +175,9 @@ pub trait LineMetaMut: LineMeta {
     fn set_labels(&mut self, labels: KeyValueMap) -> Result<(), LineMetaError>;
     fn set_level(&mut self, level: String) -> Result<(), LineMetaError>;
     fn set_meta(&mut self, meta: Value) -> Result<(), LineMetaError>;
+    fn get_line(&mut self) -> (Option<&str>, Option<&[u8]>);
+    fn set_line_text(&mut self, line: String) -> Result<(), LineMetaError>;
+    fn set_line_buffer(&mut self, line: Vec<u8>) -> Result<(), LineMetaError>;
```

Comment:
The set methods could be collapsed into a single method that takes a &[u8] or &str and returns a Result<(), LineMetaError>, then it would be the implementation's job to (optionally) validate the bytes and mutate the internal string or Vec.

This might be an additional string validation for the linebuilder types, but they are already the alloc/copy-heavy path so I don't think it would make much difference

Reply:
Makes sense, I'll change it.